File: shell-examples/src/main/js/io/vertx/example/shell/echokeyboard/echo_keyboard_command.js
var CommandBuilder = require("vertx-shell-js/command_builder");
var ShellService = require("vertx-shell-js/shell_service");
var CommandRegistry = require("vertx-shell-js/command_registry");

var starwars = CommandBuilder.command("echokeyboard").processHandler(function (process) {

  // Echo
  process.setStdin(function (keys) {
    process.write(keys.replace('\r', '\n'));
  });

  // Terminate when user hits Ctrl-C
  process.interruptHandler(function (v) {
    process.end();
  });

}).build(vertx);

var service = ShellService.create(vertx, {
  "telnetOptions" : {
    "host" : "localhost",
    "port" : 3000
  }
});
CommandRegistry.getShared(vertx).registerCommand(starwars);
service.start(function (ar, ar_err) {
  if (!ar_err == null) {
    ar_err.printStackTrace();
  }
});

File: shell-examples/src/main/ruby/io/vertx/example/shell/echokeyboard/echo_keyboard_command.rb
require 'vertx-shell/command_builder'
require 'vertx-shell/shell_service'
require 'vertx-shell/command_registry'

starwars = VertxShell::CommandBuilder.command("echokeyboard").process_handler() { |process|

  # Echo
  process.stdin_handler() { |keys|
    process.write(keys.replace('\r', '\n'))
  }

  # Terminate when user hits Ctrl-C
  process.interrupt_handler() { |v|
    process.end()
  }

}.build($vertx)

service = VertxShell::ShellService.create($vertx, {
  'telnetOptions' => {
    'host' => "localhost",
    'port' => 3000
  }
})
VertxShell::CommandRegistry.get_shared($vertx).register_command(starwars)
service.start() { |ar_err,ar|
  if (!ar_err == nil)
    ar_err.print_stack_trace()
  end
}
